Sources/APIProvider.swift

```diff
@@ -135,28 +122,38 @@ extension DataRequest {
     ///   - type: The type to map to.
     ///   - completion: The result of the mapping. An error will be returned if mapping fails.
     @discardableResult
-    func mapResponseTo<T: Decodable>(_ type: T.Type, decoder: JSONDecoder, completion: @escaping RequestCompletionHandler<T>) -> Self {
-        return responseData(queue: DispatchQueue.global(qos: .background)) { (response) in
-            if let error = response.error {
+    func dataResponse(decoder: JSONDecoder, completion: @escaping RequestCompletionHandler<Data?>) -> Self {
+        return validate(statusCode: 200..<300).responseData(queue: DispatchQueue.global(qos: .background)) {
+            if let error = $0.error {
                 // Try to parse api error
-                guard let data = response.data, let apiError = try? decoder.decode(ErrorResponse.self, from: data) else {
+                guard let data = $0.data, let apiError = try? decoder.decode(ErrorResponse.self, from: data) else {
```

Comment:
I would keep response here for readability!

Reply:
I agree, done :)